cola/widgets/branch.py

```diff
@@ -0,0 +1,520 @@
+                                                SEPARATOR_CHAR)
+        remote_branch = gitcmds.tracked_branch(branch)
+
+        if remote_branch is not None:
+            rgx = re.compile(r'^(?P<remote>[^/]+)/(?P<branch>.+)$')
+            match = rgx.match(remote_branch)
+
+            if match:
+                remote = match.group('remote')
+                branch_name = match.group('branch')
+                self.git_action_async('fetch', [remote, branch_name])
+
+    def push_action(self):
+        branch = self.tree_helper.get_full_name(self.selected_item(),
+                                                SEPARATOR_CHAR)
+        remote_branch = gitcmds.tracked_branch(branch)
+
+        if remote_branch is not None:
+            rgx = re.compile(r'^(?P<remote>[^/]+)/(?P<branch>.+)$')
+            match = rgx.match(remote_branch)
+
+            if match:
+                remote = match.group('remote')
+                branch_name = match.group('branch')
+                self.git_action_async('push', [remote, branch_name])
+
+    def rename_action(self):
+        branch = self.tree_helper.get_full_name(self.selected_item(),
+                                                SEPARATOR_CHAR)
+        new_branch = qtutils.prompt(N_("Rename branch"),
+                                    N_("Enter New Branch Name"),
+                                    branch)
+        if new_branch[1] is True and new_branch[0]:
+            self.git_action_async('rename', [branch, new_branch[0]],
+                                  refresh_tree=True)
+
+    def pull_action(self):
+        branch = self.tree_helper.get_full_name(self.selected_item(),
+                                                SEPARATOR_CHAR)
+        remote_branch = gitcmds.tracked_branch(branch)
+
+        if remote_branch is not None:
+            rgx = re.compile(r'^(?P<remote>[^/]+)/(?P<branch>.+)$')
+            match = rgx.match(remote_branch)
+
+            if match:
+                remote = match.group('remote')
+                branch_name = match.group('branch')
+                self.git_action_async('pull', [remote, branch_name],
+                                      refresh_tree=True)
+
+    def delete_action(self):
+        title = N_('Delete Branch')
+        question = N_('Delete selected branch?')
+        info = N_('The branch will be no longer available.')
+        ok_btn = N_('Delete Branch')
+
+        branch = self.tree_helper.get_full_name(self.selected_item(),
+                                                SEPARATOR_CHAR)
+
+        if branch != self.current_branch and qtutils.confirm(title, question,
+                                                             info, ok_btn):
+            remote = False
+            root = self.tree_helper.get_root(self.selected_item())
+            if NAME_REMOTE_BRANCH == root.name:
+                remote = True
+
+            if remote is False:
+                self.git_action_async('delete_local', [branch])
```

Comment:
In the conditional above, instead of saying `if remote is False:`, in python it's more idiomatic to write `if not remote:`.  Not necessary, but flipping the flow so that it's a positive test is preferred, e.g. do the `if remote:` ... part first, then do `not remote` part in the `else:`.